### ISSUE TITLE
Directory or file scanning crashes retroarch

### ISSUE THREAD

**Issue by bronnel** [libretro/RetroArch#2407]
The latet nightly builds crashes retroarch during adding content whether you scan directory or file, I have iOS and android it happens on both of them maybe other platform too.

**Comment by fr500** [comment]
it doesn't happen on android for me
Make sure you first go to online updater and update the databases

**Comment by bronnel** [comment]
I updated the database, reinstalled retroarch, double checked all directories still crashes on both iOS and android.

**Comment by twinaphex** [comment]
Tell me if you can compile from the commandline, otherwise you cannot perform the necessary test to narrow down this platform.

**Comment by brujo5** [comment]
this was the latest build working for me http://buildbot.libretro.com/nightly/apple/ios9/2015-11-14_RetroArc_iOS9.deb
all builds from 2015/11/15 and 2015/11/16 crash at load any rom for all cores

**Comment by fr500** [comment]
@brujo5 por que haces esto? por que entras a cualquier conversacion que no tiene nada que ver con tu problema a hablar de tu problema? si hay un problema, abre un issue y cierralo cuando se solucione.

**Comment by twinaphex** [comment]
It could be the HAVE_MMAP defines that were added, try taking that out for building and try again.

**Comment by brujo5** [comment]
@fr500 
mencionan ios y yo no tengo este problema de crasheo al escanear directorio para contenido. -aclarando-

al contrario ami me crashea al querer cargar cualquier rom. otro usario en el foro tambien lo menciona.

**Comment by fr500** [comment]
@brujo5 por eso, tu problema no tiene NADA que ver con este, como dije antes, no es un foro, y comentarios que no tienen nada que ver solo hacen mas dificil dar solucion al incidente original.

**Comment by brujo5** [comment]
esta bien pero creo que es lo mismo su crasheo y el que yo experimento tienen el mismo culpable. pero esta bien deja y borro los mensajes.

**Comment by bronnel** [comment]
Unfortunately I can't , I wish I could

**Comment by twinaphex** [comment]
Pushed a commit that disables HAVE_MMAP for both iOS/Android, if the next nightly works again then it was down to that.

**Comment by fr500** [comment]
@bronnel what device do you have (android) 
I find it weird, I just scanned my snes set and it seems perfectly fine

**Comment by bronnel** [comment]
I Have sony xperia z5, and iPad mini 2 ios 8.4

**Comment by bronnel** [comment]
You are right, I just tested it only happens with iso files

**Comment by bronnel** [comment]
Yes and double checked this only happens with ISO files,

**Comment by bronnel** [comment]
I tested with NES SNES N64 alll working fine but with iso files of psx, it crashes

**Comment by bronnel** [comment]
Also GBA working fine

**Comment by twinaphex** [comment]
Can you wait until the next nightly? I just pushed that commit, I dunno when the buildbot is going to be done churning out the new builds but it should be soonish.

**Comment by bronnel** [comment]
Ok thanks a lot, I'm waiting to it

**Comment by bronnel** [comment]
I downloaded and installed the latest nightly just now still the problem persists. Like I said it only happens with ISO files.

**Comment by twinaphex** [comment]
Which core? When did it last work? Start being specific or we cannot fix these issues.

**Comment by bronnel** [comment]
Ok when you add content by [ Scan file ] for psx files (cue, bin) and psp files (iso) the last working build for this was 2015-11-13_RetroArch.deb
From 2015-11-14_RetroArch.deb it crashes retroarch 

Note :- adding content by [ Scan directory] for psx files ( bin, cue) was always problematic from the early builds until now, 
But adding content by  [ Scan directory] for psp files (iso) was working until 2015-11-13   (Being the last working build)

I tested this very accurately on my iPad mini 2 running iOS 8.4 with Taij jailbreak
Also tested on my Xperia Z5 running android 5.1.1

**Comment by bronnel** [comment]
Ok when you add content by [ Scan file ] for psx core (pcsx rearmead) files (cue, bin) and ppsspp files (iso) the last working build for this was 2015-11-13_RetroArch.deb
From 2015-11-14_RetroArch.deb it crashes retroarch

Note :- adding content by [ Scan directory] on iOS for psx files ( bin, cue) was always problematic from the early builds until now 
But adding content by [ Scan directory] for psp files (iso) was working until 2015-11-13 (Being the last working build)

tested this very accurately on my iPad mini 2 running iOS 8.4 with Taij jailbreak
Also tested on my Xperia Z5 running android 5.1.1

Bottom line is that the last working build for adding content for [ iso, bin, cue] was Nov 13 2015 , On android both (scan directory and scan file) was working, on iOS however only scan file was working

**Comment by twinaphex** [comment]
@heuripedes added most of the changes to scanning during that time, I'm not really sure what could have caused it.

**Comment by heuripedes** [comment]
Sent some fixes to the content scanning code, should be available on today's nightly.

**Comment by bronnel** [comment]
https://github.com/heuripedes I can confirm that the problem has been fixed,

**Comment by bronnel** [comment]
I can now confirm that the issues has been solved thanks to you guys, great job.
